### PR TITLE
Enable passing custom calico version

### PR DESCRIPTION
**Pull request by jsturtevant**
**Description**
Use the calico environment variable to install custom helm version. This helps prevents against calico from auto upgrading.

**History Notes**

---

This checklist is used to make sure that common guidelines for an Azure CLI pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).
- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).
- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/capi/azext_capi/custom.py [hunk 1]
```diff
@@ -618,7 +618,7 @@ def install_cni(cmd, cluster_name, workload_cfg, windows, args):
         chart_name="calico",
         chart="projectcalico/tigera-operator",
         values_file=values_file,
-        args=["--namespace", "tigera-operator", "--create-namespace"]
+        args=["--namespace", "tigera-operator", "--create-namespace", "--version",  os.environ.get("CALICO_VERSION", "v3.25.1")]
```
Comment (marosset) [inline src/capi/azext_capi/custom.py:621]: I think we need to update line 642 below to also use `release-1.9` for the windows calico deployments

Comment (marosset) [reply src/capi/azext_capi/custom.py:621]: And should we do something like 
 
```bash
kubectl set image -n calico-system daemonSet/calico-node-windows install-cni sigwindowstools/calico-install:${CALICO_VERSION}-hostprocess
kubectl set image -n calico-system daemonSet/calico-node-windows calico-node-startup sigwindowstools/calico-node:${CALICO_VERSION}-hostprocess
kubectl set image -n calico-system daemonSet/calico-node-windows calico-node-felix sigwindowstools/calico-node:${CALICO_VERSION}-hostprocess
```
in `install_cni_windows` to make sure we get the same version?

**Review by marosset — DISMISSED** [review]
/lgtm

thanks for the updates!

**Comment by jsturtevant** [comment]
validating with https://github.com/kubernetes-sigs/windows-testing/pull/380

File: src/capi/azext_capi/helpers/constants.py [hunk 2]
```diff
@@ -7,3 +7,5 @@
 KUBECONFIG = "KUBECONFIG"
 AKS_INFRA_RG_NAME = "AKS_INFRA_RG_NAME"
 AKS_VNET_NAME = "AKS_VNET_NAME"
+CAPZ_BASE_CONTENT_URL = "https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-azure/release-1.9"
```
Comment (CecileRobertMichon) [inline src/capi/azext_capi/helpers/constants.py:10]: this is bumping capz to release-1.9 in the process (we were using 1.7 before) which might break other things (specifically around the out of tree switch), would be safer to make that change separately

Comment (jsturtevant) [reply src/capi/azext_capi/helpers/constants.py]: moved back to 1.7

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/assign @mboersma

**Review by CecileRobertMichon — APPROVED** [review]
/lgtm
/assign @mboersma

**Review by mboersma — COMMENTED** [review]
This looks good, thank you! Could you add a bullet point to `src/capi/HISTORY.rst` since this is user-visible?

**Comment by jsturtevant** [comment]
hmmm, my test locally is failing with 
```
error: there is no need to specify a resource type as a separate argument when passing arguments in resource/name form (e.g. 'kubectl get resource/<resource_name>' instead of 'kubectl get resource resource/<resource_name>'

cli: Command '['kubectl', 'set', 'image', '-n ', 'calico-system', 'daemonSet/calico-node-windows', 
'install-cni=sigwindowstools/calico-install:v3.25.1-hostprocess', 
'--kubeconfig', 'test-win1.kubeconfig']' returned non-zero exit status 1.
```
but when I run that command manually it works:

```
 kubectl set image -n calico-system daemonSet/calico-node-windows install-cni=sigwindowstools/calico-install:v3.25.1-hostprocess --kubeconfig test-win1.kubeconfig
daemonset.apps/calico-node-windows image updated
```

What am I missing?

**Comment by mboersma** [comment]
That's weird, because you're not passing resource type as a separate argument AFAICT. Hmm...

**Comment by jsturtevant** [comment]
It appears to be the extra space in ` '-n ',`  🤷.  Running a test now.

**Comment by mboersma** [comment]
That's weird that whitespace would be interpreted there as an extra argument, but I bet you're right. I tried `az capi create -l westus3 --debug` locally with this branch and didn't see any errors: workload cluster came up fine.

**Review by mboersma — DISMISSED** [review]
/lgtm

@jsturtevant feel free to merge this when you're comfortable with your testing.

**Comment by jsturtevant** [comment]
It was indeed the extra space, I also had to get the container name and images correct but we have a passing job: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_windows-testing/380/pull-e2e-capz-windows-2022-extension/1664370802792861696

**Pull request merged**